src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs

```diff
@@ -0,0 +1,332 @@
+
+        var optionsProvider = new TestAnalyzerConfigOptionsProvider();
+        optionsProvider.TestGlobalOptions["build_property.RazorConfiguration"] = "Default";
+        optionsProvider.TestGlobalOptions["build_property.RootNamespace"] = "MyApp";
+        optionsProvider.TestGlobalOptions["build_property.RazorLangVersion"] = "Latest";
+        optionsProvider.TestGlobalOptions["build_property.GenerateRazorMetadataSourceChecksumAttributes"] = "false";
+
+        var additionalTexts = ImmutableArray<AdditionalText>.Empty;
+
+        foreach (var document in project.AdditionalDocuments)
+        {
+            var additionalText = new TestAdditionalText(document.Name, await document.GetTextAsync());
+            additionalTexts = additionalTexts.Add(additionalText);
+
+            var additionalTextOptions = new TestAnalyzerConfigOptions
+            {
+                ["build_metadata.AdditionalFiles.TargetPath"] = Convert.ToBase64String(Encoding.UTF8.GetBytes(additionalText.Path)),
+            };
+
+            optionsProvider.AdditionalTextOptions[additionalText.Path] = additionalTextOptions;
+        }
+
+        configureGlobalOptions?.Invoke(optionsProvider);
+
+        driver = driver
+            .AddAdditionalTexts(additionalTexts)
+            .WithUpdatedAnalyzerConfigOptions(optionsProvider);
+
+        return (driver, additionalTexts, optionsProvider);
+    }
+
+    protected static GeneratorRunResult RunGenerator(Compilation compilation, ref GeneratorDriver driver, params Action<Diagnostic>[] expectedDiagnostics)
+    {
+        driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out compilation, out _);
+
+        var actualDiagnostics = compilation.GetDiagnostics().Where(d => d.Severity != DiagnosticSeverity.Hidden);
+        Assert.Collection(actualDiagnostics, expectedDiagnostics);
+
+        var result = driver.GetRunResult();
+        return result.Results.Single();
+    }
+
+    protected static Project CreateTestProject(
+        Dictionary<string, string> additonalSources,
+        Dictionary<string, string>? sources = null)
+    {
+        var project = CreateBaseProject();
+
+        if (sources is not null)
+        {
+            foreach (var (name, source) in sources)
+            {
+                project = project.AddDocument(name, source).Project;
+            }
+        }
+
+        foreach (var (name, source) in additonalSources)
+        {
+            project = project.AddAdditionalDocument(name, source).Project;
+        }
+
+        return project;
+    }
+
+    private class AppLocalResolver : ICompilationAssemblyResolver
+    {
+        public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string>? assemblies)
+        {
+            foreach (var assembly in library.Assemblies)
+            {
+                var dll = Path.Combine(Directory.GetCurrentDirectory(), "refs", Path.GetFileName(assembly));
```

Comment:
APIs like `Directory.GetCurrentDirectory` represent process wide state. This means the test outcome can be changed by this value being changed in other parts of the test. Rather than use this ambient state here, make it a parameter to `AppLocalResolver` and force callers to provide the correct directory. 

Yes they may put `GetCurrentDirectory` into the test itself but that makes it much more obvious that the test is using ambient state. Here it's hidden.

Reply:
Added a parameter. There's only one caller - `CreateBaseProject` below. I used as the argument `AppContext.BaseDirectory` instead of `Directory.GetCurrentDirectory()`, seems better.

Reply:
Yeah that is much better.